### PR TITLE
cope with recent pymavlink string changes

### PR DESCRIPTION
**Pull request by tridge**
ping @alehed

### CODE REVIEW

File: MAVProxy/modules/mavproxy_swarm.py [hunk 1]
```diff
@@ -939,7 +939,7 @@ def mavlink_packet(self, m):
                 self.vehicleLastHB[(sysid, compid)] = time.time()
 
             # updated leader information from vehicle
-            elif mtype == 'PARAM_VALUE' and m.param_id == "FOLL_SYSID":
+            elif mtype == 'PARAM_VALUE' and m.param_id.decode('ascii') == "FOLL_SYSID":
```
Comment (alehed) [inline MAVProxy/modules/mavproxy_swarm.py:942]: Here I would compare to a bytes literal.

**Review by alehed — COMMENTED** [review]
Looks good. I made one comment about avoiding a decode call. Does mavproxy specify which version of pymavlink to use with it?

If not, we probably need to stay compatible with older versions of pymavlink, in which case param_id can also be a string. Calling decode on a string fails, so in that case there would have to be a type check in the code.

**Comment by alehed** [comment]
There are probably more changes needed. For instance in [MAVProxy/modules/mavproxy_ais.py‎](https://cs.github.com/ArduPilot/MAVProxy/blob/22ffc05170df048b9465ae2f664d272e19209b22/MAVProxy/modules/mavproxy_ais.py?q=owner%3AArduPilot+language%3Apython+callsign) on line 102, there is a str() which should be a decode. I will try to find a way to exhaustively find these cases.

**Comment by tridge** [comment]
not needed any more

**Pull request closed without merging**